snuba/utils/streams/backends/kafka.py

```diff
@@ -695,6 +698,14 @@ def get_default_kafka_configuration(
 
     if override_params:
         broker_config.update(override_params)
+
+    broker_config["log_level"] = pylog_to_syslog_level(logger.getEffectiveLevel())
+    # Without this, rdkafka simply blurts log lines out into stdout/stderr
+    broker_config["log.queue"] = "true"
```

Comment:
@fpacifici had to add this back as we see the transitional errors on stdout without this.

Reply:
I think this is still a problem. 
From the documentation it seems that data is flushed to the log queue only upon `poll` and `flush`, which means we would be missing issues during the consumer creation and subscription. Including those errors during to legit connection errors:

> To avoid spontaneous calls from non-Python threads the log messages will only be forwarded when client.poll() or producer.flush() are called.


Example. Here I tried to connect to Kafka on the wrong port:
With log.queue=True
```
(snuba) ~/snuba  (feat/continuous_bulk_stream *)$ snuba consumer --storage groupedmessages --auto-offset-reset earliest  --log-level DEBUG --consumer-group cdc_group3
2021-03-16 14:05:40,210 Starting
^C
2021-03-16 14:06:08,271 Shutdown signalled
2021-03-16 14:06:08,272 Stopping consumer
2021-03-16 14:06:08,275 Stopped
```

Without:
```
(snuba) ~/snuba  (feat/continuous_bulk_stream *)$ snuba consumer --storage groupedmessages --auto-offset-reset earliest  --log-level DEBUG --consumer-group cdc_group3
2021-03-16 14:06:14,383 Starting
%3|1615928774.384|FAIL|rdkafka#consumer-2| [thrd:localhost:9094/bootstrap]: localhost:9094/bootstrap: Connect to ipv4#127.0.0.1:9094 failed: Connection refused (after 1ms in state CONNECT)
%3|1615928774.384|FAIL|rdkafka#producer-1| [thrd:localhost:9094/bootstrap]: localhost:9094/bootstrap: Connect to ipv6#[::1]:9094 failed: Connection refused (after 1ms in state CONNECT)
%3|1615928775.387|FAIL|rdkafka#consumer-2| [thrd:localhost:9094/bootstrap]: localhost:9094/bootstrap: Connect to ipv6#[::1]:9094 failed: Connection refused (after 0ms in state CONNECT)
%3|1615928775.387|FAIL|rdkafka#producer-1| [thrd:localhost:9094/bootstrap]: localhost:9094/bootstrap: Connect to ipv6#[::1]:9094 failed: Connection refused (after 0ms in state CONNECT, 1 identical error(s) suppressed)
^C
2021-03-16 14:06:16,388 Shutdown signalled
2021-03-16 14:06:16,388 Stopping consumer
%3|1615928776.390|FAIL|rdkafka#producer-1| [thrd:localhost:9094/bootstrap]: localhost:9094/bootstrap: Connect to ipv4#127.0.0.1:9094 failed: Connection refused (after 1ms in state CONNECT)
2021-03-16 14:06:16,390 Stopped
```

So we would not have any indication of what went wrong by simply adding this flag. I am not sure if there is a better way to track error messages during assignment and connection, I can look into that, though it seems that simply adding this flag may be not ehough.

Reply:
By the way, this settings can be added to settings.py if some user needs to hide those logs even with the added cost.
We would have to add those properties here to make it work: https://github.com/getsentry/snuba/blob/master/snuba/utils/streams/backends/kafka.py#L626-L648 bt that is fine.